4D/matRad_makeBixelTimeSeq.m

```diff
@@ -0,0 +1,202 @@
+function bixelInfo = matRad_makeBixelTimeSeq(stf, resultGUI)
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% using the steering information of matRad, makes a time sequenced order
+% according to the irradiation scheme in spot scanning
+%
+% call
+%   bixelInfo = matRad_makeBixelTimeSeq(stf, resultGUI)
+%
+% input
+%   stf:            matRad steering information struct
+%   resultGUI:      struct containing optimized fluence vector
+%
+% output
+%   bixelInfo:      struct containing bixel ordering information and the
+%                   time sequence of the spot scanning
+%
+% References
+%   spill structure and timing informations:
+%   http://cdsweb.cern.ch/record/1182954
+%   http://iopscience.iop.org/article/10.1088/0031-9155/56/20/003/meta
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%
+% Copyright 2018 the matRad development team.
+% 
+% This file is part of the matRad project. It is subject to the license 
+% terms in the LICENSE file found in the top-level directory of this 
+% distribution and at https://github.com/e0404/matRad/LICENSES.txt. No part 
+% of the matRad project, including this file, may be copied, modified, 
+% propagated, or distributed except according to the terms contained in the 
+% LICENSE file.
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+if nargin < 3
+    plotting = 'off';
+end
+
+% defining the constant parameters
+%
+% time required for synchrotron to change energy
+es_time = 4 * 10^6;
+% time required for synchrotron to recharge it's spill
+spill_recharge_time = 2 * 10^6;
+% number of particles generated in each spill
+spill_size = 4 * 10 ^ 10;
+% speed of synchrotron's lateral scanning in an IES
+scan_speed = 10; % m/s
+% number of particles per second
+spill_intensity = 4 * 10 ^ 8;
+
+
+for i = 1:length(stf)
+    steerTime(i) = stf(i).bixelWidth * (10 ^ 3)/ scan_speed;
```

Comment:
@ahmadnish when you work with a fixed time doesnt this mean that you do neglect the actual distance between two spots in one iso energy slice?

Reply:
But isn't the actual distance between two spots is always the bixelWidth, unless you mean when it changes rows, which in that case, yeah you are right I'll be neglecting that here. I'll see if that can be implemented